File: typescript/src/renderer.ts
import {Shape} from "./shapes/shape.js";

function Renderer(shape: Shape) {
    return {
        draw() {
            const area: number = shape.computeArea()
            console.log("Shape drawn\n" + "Its area is " + area)
        }
    }
}

function newRenderer(shape: Shape) {
    return {
        draw() {
            const area: number = shape.computeArea()
            console.log("Shape drawn\n" + "Its area is " + area)
        }
    }
}

export { Renderer,newRenderer}

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer {
    private Shape _shape;
    
    Renderer(Shape shape) {
        this._shape = shape;
    }

    void draw() {
        double area = _shape.getArea();
        // assume implementation
        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
